uberenv.py

```diff
@@ -1268,29 +1238,32 @@ def main():
     # Setup the necessary paths and directories
     env.setup_paths_and_dirs()
 
-    # Skip creating and setting up environment if env given 
+    # Go to package manager's destination
+    os.chdir(env.dest_dir)
+
+    # Setup package manager
     if not opts["skip_setup"]:
         # Clone the package manager
         env.clone_repo()
 
-        os.chdir(env.dest_dir)
-
         # Patch the package manager, as necessary
         env.patch()
 
         # Clean the build
         env.clean_build()
 
-        # Allow to end uberenv after spack is ready
+        # Allow to end uberenv after Spack is ready
         if opts["setup_only"]:
 
+            # Use Spack upstream
             if not is_windows() and opts["upstream"] is not None:
                 env.use_spack_upstream()
 
             return 0
-    else:
-        os.chdir(env.dest_dir)
 
+    # Create Spack Environment
+    if not is_windows():
+        env.create_spack_env()
```

Comment:
In "create_spack_env" the environment is also concretized. It means that it’s concretized before the mirrors and upstreams are set, so the concretization cannot use them. Both could contain already installed packages, especially when using `--reuse`.

Setting up the mirrors and upstreams must also be done before looking for already installed spec in show info, for the same reason.

In the end, I think both concretization and checking for already installed package should be done right before the install. That does not apply to "creating the environment" nor "showing information" by the way.

Reply:
So you're saying setting up mirrors and upstreams should be done first, and then concretize?

Reply:
Yes, move concretize after, or move mirrors and upstreams before, that’s the same :)

Reply:
should be fixed in the following commit

@white238 thoughts on creating a mirror CI check in the future?